shared/actions/unlock-folders.js

```diff
@@ -38,10 +39,23 @@ export function toPaperKeyInput (): ToPaperKeyInput {
   return {type: Constants.toPaperKeyInput, payload: {}}
 }
 
-export function checkPaperKey (): TypedAsyncAction<CheckPaperKey> {
+export function onBackFromPaperKey (): OnBackFromPaperKey {
+  return {type: Constants.onBackFromPaperKey, payload: {}}
+}
+
+function waiting (currentlyWaiting: boolean): Waiting {
+  return {type: Constants.waiting, payload: currentlyWaiting}
+}
+
+export function checkPaperKey (paperKey: HiddenString): TypedAsyncAction<CheckPaperKey | Waiting> {
   return dispatch => {
     // TODO figure out what service request to ask for
-    dispatch({type: Constants.checkPaperKey, payload: {success: true}})
+
+    dispatch(waiting(true))
```

Comment:
i'd just write a comment to use the waiting functionality of the engine and not keep this